ledger-api/grpc-definitions/com/daml/ledger/api/v1/command_submission_service.proto

```diff
@@ -37,11 +37,17 @@ service CommandSubmissionService {
   // Errors:
   // - ``UNAUTHENTICATED``: if the request does not include a valid access token
   // - ``PERMISSION_DENIED``: if the claims in the token are insufficient to perform a given operation
-  // - ``NOT_FOUND``: if the request does not include a valid ledger id
+  // - ``NOT_FOUND``: if the request does not include a valid ledger id or the submission rank or
+  //   deduplication period are too old.
   // - ``INVALID_ARGUMENT``: if the payload is malformed or is missing required fields
   // - ``UNAVAILABLE``: if the participant is not yet ready to submit commands or if the service has been shut down.
   // - ``RESOURCE_EXHAUSTED``: if the participant or the ledger is overloaded. Clients should back off exponentially and retry.
-  // - ``ABORTED``: if a contract key is missing or duplicated due to for example contention on resources
+  // - ``ABORTED``: if a contract key is missing or duplicated due to for example contention on resources or
+  //   if there is another submission in flight for the same change ID.
+  // - ``FAILED_PRECONDITION`` if there is another submission with a higher submission rank.
```

Comment:
@andreaslochbihler-da : this is a definite rejection, that I'd expect to be communicated via a completion. Is that so?

Reply:
This way the caller is guaranteed to get a new completion offset to use a the submission rank.

Reply:
If there's a conflict on ranks, we agreed to make that a cancellation. We should have here the same text as for the command service, namely that the error details contain the conflicting rank.

We should find a way to avoid copy-pasting all this stuff.